### PR TITLE
Add config map backend (list and details)

### PR DESCRIPTION
**Pull request by bryk**
Addresses #1011

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/dashboard/1012)

### CODE REVIEW

**Comment by bryk** [comment]
@floreks Can you review this? This now includes your commit, but I'll rebase once you're merged.

File: src/app/backend/resource/configmap/configmapdetail.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"log"
+
+	"github.com/kubernetes/dashboard/src/app/backend/resource/common"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// The ConfigMap API resource provides mechanisms to inject containers with configuration data while keeping
```
Comment (floreks) [inline src/app/backend/resource/configmap/configmapdetail.go]: Start with struct name to make golint happy?

File: src/app/backend/resource/configmap/configmaplist.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+// ConfigMapList contains a list of Config Maps in the cluster.
+type ConfigMapList struct {
+	ListMeta common.ListMeta `json:"listMeta"`
+
+	// Unordered list of Config Maps
+	Items []ConfigMap `json:"items"`
+}
+
+// The ConfigMap API resource provides mechanisms to inject containers with configuration data while keeping
```
Comment (floreks) [inline src/app/backend/resource/configmap/configmaplist.go]: Same here

**Comment by edouardKaiser** [comment]
Love the feature, can't wait to be able to use it.
With this, most of the time, I could avoid using kubectl.

**Comment by bryk** [comment]
PTAL

---

Review status: 0 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/app/backend/resource/configmap/configmapdetail.go, line 25 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/1012#-KMTecRdMzz9MJOkau9f-r2-25:-KMUD_Gsl29avXH7Ipy5:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/22c0e9b82f6966f2270fce7b6a702d266f163a61/src/app/backend/resource/configmap/configmapdetail.go#L25)):*
<details><summary><i>Previously, floreks (Sebastian Florek) wrote…</i></summary>
> Start with struct name to make golint happy?
</details>
Done.

---

*[src/app/backend/resource/configmap/configmaplist.go, line 33 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/1012#-KMTecRdMzz9MJOkau9g-r2-33:-KMUDcpmufpnNY5DcTLs:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/22c0e9b82f6966f2270fce7b6a702d266f163a61/src/app/backend/resource/configmap/configmaplist.go#L33)):*
<details><summary><i>Previously, floreks (Sebastian Florek) wrote…</i></summary>
> Same here
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/1012#-:-KMUDiogeLUZo9iBwU8P:676965300)*

**Comment by floreks** [comment]
:lgtm:

---

Reviewed 5 of 7 files at r2, 2 of 7 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/1012#-:-KMUEubPZwmcLT8yn63r:1417088721)*

**Pull request merged**